astropy/units/quantity_helper/function_helpers.py

```diff
@@ -497,6 +501,63 @@ def arange_impl(*args, start=None, stop=None, step=None, dtype=None, device=None
     return np.arange(**kwargs), out_unit, None
 
 
+if NUMPY_LT_2_0:
+    @dispatched_function
+    def empty(shape, dtype=None, order='C'):
+        return empty_impl(shape, dtype, order)
+else:
+    @dispatched_function
+    def empty(shape, dtype=None, order='C', *, device=None):
+        return empty_impl(shape, dtype, order, device=device)
+
+def empty_impl(shape, dtype=None, order='C', *, device=None):
+    if NUMPY_LT_2_0:
+        ret_arr = np.empty(shape, dtype, order)
+    else:
+        ret_arr = np.empty(shape, dtype, order, device=device)
+    return ret_arr, _UnitFromLikeArg, None
+
+
+if NUMPY_LT_2_0:
+    @dispatched_function
+    def ones(shape, dtype=None, order='C'):
+        return ones_impl(shape, dtype, order)
+else:
+    @dispatched_function
+    def ones(shape, dtype=None, order='C', *, device=None):
+        return ones_impl(shape, dtype, order, device=device)
+
+def ones_impl(shape, dtype=None, order='C', *, device=None):
```

Comment:
Same as above. Alternatively, one could have an implementation for all those creation functions (i.e., also pass a `function=np.ones` along).